src/init.c

```diff
@@ -869,9 +870,10 @@ static NOINLINE void _finish_julia_init(JL_IMAGE_SEARCH rel, jl_ptls_t ptls, jl_
     JL_TIMING(JULIA_INIT, JULIA_INIT);
     jl_resolve_sysimg_location(rel);
     // loads sysimg if available, and conditionally sets jl_options.cpu_target
+    // If the image is in memory then jl_base_image_handle must be set before calling init
     if (rel == JL_IMAGE_IN_MEMORY) {
-        jl_set_sysimg_so(jl_exe_handle);
-        jl_options.image_file = jl_options.julia_bin;
+        jl_set_sysimg_so(jl_base_image_handle);
+        jl_options.image_file = jl_options.julia_bin; // XXX: Use dladdr on base_image_handle?
```

Comment:
It is possible to pass this in explicitly instead of relying on it to be initialized externally?